Userland/Libraries/LibGUI/Application.cpp

```diff
@@ -324,4 +325,73 @@ void Application::event(Core::Event& event)
     Object::event(event);
 }
 
+void Application::set_config_domain(String config_domain)
+{
+    m_config_domain = move(config_domain);
+}
+
+void Application::register_recent_file_actions(Badge<GUI::Menu>, Vector<NonnullRefPtr<GUI::Action>> actions)
+{
+    m_recent_file_actions = move(actions);
+    update_recent_file_actions();
+}
+
+void Application::update_recent_file_actions()
+{
+    size_t number_of_recently_open_files = 0;
+    auto update_action = [&](size_t index) {
+        auto& action = m_recent_file_actions[index];
+        char buffer = static_cast<char>('0' + index);
```

Comment:
Maybe add a `static_assert(max_recently_open_files() <= 9)` before we get `:` as a key?